DuckDuckGo.Fluent.Plugin/DuckDuckGoSearchApp.cs

```diff
@@ -0,0 +1,228 @@
+                            DuckResult answers = GetAnswers(root, searchedText);
+                            if (answers != null) await channel.Writer.WriteAsync(answers, cancellationToken);
+
+                            // Get Definitions if available
+                            DuckResult dictionary = GetDictionary(root, searchedText);
+                            if (dictionary != null) await channel.Writer.WriteAsync(dictionary, cancellationToken);
+
+                            // Get Abstract in Text form
+                            DuckResult abstractResult = GetAbstract(root, searchedText);
+                            if (abstractResult != null)
+                                await channel.Writer.WriteAsync(abstractResult, cancellationToken);
+
+                            // External Links associated with search like Official Website etc.
+                            if (root?.Results != null)
+                                foreach (DuckResult duckResult in root.Results.Select(variableTopic => new DuckResult
+                                {
+                                    Info = variableTopic.Text, SourceUrl = variableTopic.FirstUrl,
+                                    ResultType = "Links",
+                                    SearchedText = searchedText, SearchResultType = ResultType.SearchResult, Score = 7
+                                }))
+                                    await channel.Writer.WriteAsync(duckResult, CancellationToken.None);
+
+                            // Internal Links associated with Search.
+                            await GetRelatedSearch(root, channel, searchedText);
+                        }
+
+                        // Finish channel if task is null.
+                        else
+                        {
+                            channel.Writer.Complete();
+                        }
+                    }, cancellationToken).ContinueWith(_ => channel.Writer.Complete(), CancellationToken.None);
+
+                await foreach (DuckResult duckResult in channel.Reader.ReadAllAsync(cancellationToken))
+                    yield return new DuckDuckGoSearchResult(duckResult.Info, searchedText, duckResult.ResultType,
+                        DuckOperations, duckResult.Score)
+                    {
+                        Url = duckResult.SourceUrl, AdditionalInformation = duckResult.SourceUrl,
+                        SearchObjectId = duckResult, PreviewImage = _logoImage
+                    };
+            }
+        }
+
+        public ValueTask LoadSearchApplicationAsync()
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            const string resourceName = "DuckDuckGo.Fluent.Plugin.duck_logo.png";
+            using Stream stream = assembly.GetManifestResourceStream(resourceName);
+            var image = new Bitmap(stream!);
+            _logoImage = new BitmapImageResult(image) {ScaleX = 1.3, ScaleY = 1.3};
```

Comment:
You can pass the stream immediately (without `using`) it should be more efficient